src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp

```diff
@@ -1472,9 +1473,282 @@ namespace OpenLoco::Drawing
+                        break;
+
+                    case ControlCodes::Colour::green:
+                        setTextColour(4);
+                        break;
+
+                    case ControlCodes::Colour::yellow:
+                        setTextColour(5);
+                        break;
+
+                    case ControlCodes::Colour::topaz:
+                        setTextColour(6);
+                        break;
+
+                    case ControlCodes::Colour::celadon:
+                        setTextColour(7);
+                        break;
+
+                    case ControlCodes::Colour::babyBlue:
+                        setTextColour(8);
+                        break;
+
+                    case ControlCodes::Colour::paleLavender:
+                        setTextColour(9);
+                        break;
+
+                    case ControlCodes::Colour::paleGold:
+                        setTextColour(10);
+                        break;
+
+                    case ControlCodes::Colour::lightPink:
+                        setTextColour(11);
+                        break;
+
+                    case ControlCodes::Colour::pearlAqua:
+                        setTextColour(12);
+                        break;
+
+                    case ControlCodes::Colour::paleSilver:
+                        setTextColour(13);
+                        break;
+
+                    default:
+                        if (chr >= 32)
+                        {
+                            numChars--;
+                        }
+                        if (!offscreen)
+                        {
+                            // When offscreen in the y dimension there is no requirement to keep pos.x correct
+                            if (chr >= 32)
+                            {
+                                // Use withPrimary to set imageId flag to use the correct palette code (Colour::black is not actually used)
+                                drawImagePaletteSet(rt, pos, ImageId(1116 + chr - 32 + getCurrentFontSpriteBase()).withPrimary(Colour::black), PaletteMap::View{ _textColours }, {});
+                                pos.x += _characterWidths[chr - 32 + getCurrentFontSpriteBase()];
+                            }
+                            else
+                            {
+                                // Unhandled control code
+                                assert(false);
+                            }
+                        }
+                        break;
+                }
+            }
+
+            return numChars;
+        }
+
+        // 0x004950EF
+        static void drawStringTicker(Gfx::RenderTarget& rt, const Ui::Point& origin, string_id stringId, Colour colour, uint8_t numLinesToDisplay, uint16_t numCharactersToDisplay, uint16_t width)
+        {
+            _currentFontSpriteBase = Font::medium_bold;
+            // Setup the text colours (FIXME: This should be a separate function)
```

Comment:
is this `fixme` for now or for later?

Reply:
its for later. I've put this same comment in many of the drawString functions they will all get addressed together (hopefully)